### ISSUE TITLE
Add support for EQUALS macro to allow for easy variable substitution

### ISSUE THREAD

**Issue by harshvcs** [ampproject/amphtml#22160]
## Add EQUALS macro to allow for easier string output mapping

There is currently no support from [available macros](https://github.com/ampproject/amphtml/blob/99d8eac351a94119233295dfda497b0b31b3967f/extensions/amp-analytics/0.1/variables.js#L153-L164) to map [variable](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#variables) values that might be required by a vendor in their amp-analytics [vendor file](https://github.com/ampproject/amphtml/tree/0a84124122b516d33e8f815c4d7205ea92f0865b/extensions/amp-analytics/0.1/vendors). To achieve multiple output string mapping, most of the vendors will need something that checks for equality - say a new EQUALS macro (that returns a===b) along with the existing $IF macro. 

Assuming $EQUALS is added, then a required mapping like "A" -> 1, "B" -> 2, "C" -> 3 can be solved using:

```js
$IF($EQUALS(${variable}, A), 1, $IF($EQUALS(${variable}, B), 2, 3));
```

## Alternative

An alternative that uses the $REPLACE macro is shared [here](https://github.com/ampproject/amphtml/issues/18034#issuecomment-474083366), but in general, it is cumbersome to implement and understand.

## Additional context

This is originally discussed with @zhouyx in #18034

**Comment by zhouyx** [comment]
to @jonathantyng-amp could you take this? Thank you

**Comment by zhouyx** [comment]
Closed by #22404
